app/models/manageiq/providers/azure_stack/inventory/collector/target_collection.rb

```diff
@@ -0,0 +1,171 @@
+class ManageIQ::Providers::AzureStack::Inventory::Collector::TargetCollection < ManageIQ::Providers::AzureStack::Inventory::Collector
+  require_nested :V2018_03_01
+  require_nested :V2017_03_09
+
+  def initialize(manager, target)
+    super
+
+    parse_targets!
+    infer_related_ems_refs!
+
+    # Reset the target cache, so we can access new targets inside
+    target.manager_refs_by_association_reset
+  end
+
+  # ##################################
+  # Target collection for CloudManager
+  # ##################################
+  def resource_groups
+    return @resource_groups unless @resource_groups.nil?
+    return [] if (refs = references(:resource_groups)).blank?
+
+    @resource_groups ||= refs.map { |ems_ref| resource_group(ems_ref) }.compact
```

Comment:
We're checking for `@resource_groups.nil?` above so the `||=` isn't technically needed and you could do a straight assignment here